pkg/controller/configmap_ca_injector/controller.go

```diff
@@ -0,0 +1,183 @@
+			return shouldUpdateConfigMaps(e.MetaNew)
+		},
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			return false
+		},
+		CreateFunc: func(e event.CreateEvent) bool {
+			return shouldUpdateConfigMaps(e.Meta)
+		},
+		GenericFunc: func(e event.GenericEvent) bool {
+			return shouldUpdateConfigMaps(e.Meta)
+		},
+	}
+
+	err = c.Watch(&source.Kind{Type: &corev1.ConfigMap{}}, &handler.EnqueueRequestForObject{}, pred)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+var _ reconcile.Reconciler = &ReconcileConfigMapInjector{}
+
+type ReconcileConfigMapInjector struct {
+	client client.Client
+	scheme *runtime.Scheme
+	status *statusmanager.StatusManager
+}
+
+// Reconcile expectes requests to refers to configmaps of two different types.
+// 1. a configmap named trusted-ca-bundle in namespace openshift-config-managed and will ensure that all configmaps with the label
+// config.openshift.io/inject-trusted-cabundle = true have the certificate information stored in trusted-ca-bundle's ca-bundle.crt entry.
+// 2. a configmap in any namespace with the label config.openshift.io/inject-trusted-cabundle = true and will insure that it contains the ca-bundle.crt
+// entry in the configmap named trusted-ca-bundle in namespace openshift-config-managed.
+func (r *ReconcileConfigMapInjector) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	log.Printf("Reconciling configmap from  %s/%s\n", request.Name, request.Namespace)
+
+	trustedCAbundleConfigMap := &corev1.ConfigMap{}
+	trustedCAbundleConfigMapName := types.NamespacedName{
+		Namespace: names.TRUSTED_CA_BUNDLE_CONFIGMAP_NS,
+		Name:      names.TRUSTED_CA_BUNDLE_CONFIGMAP,
+	}
+	err := r.client.Get(context.TODO(), trustedCAbundleConfigMapName, trustedCAbundleConfigMap)
+	if err != nil {
+		log.Println(err)
+		return reconcile.Result{}, err
+	}
+	_, trustedCAbundleData, err := validation.TrustBundleConfigMap(trustedCAbundleConfigMap)
+
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	//build a list of configMaps
+	configMapsToChange := []corev1.ConfigMap{}
+
+	//The trusted-ca-bundle changed
+	if request.Name == names.TRUSTED_CA_BUNDLE_CONFIGMAP && request.Namespace == names.TRUSTED_CA_BUNDLE_CONFIGMAP_NS {
+
+		configMapList := &corev1.ConfigMapList{}
+		selector := labels.Set(map[string]string{names.TRUSTED_CA_BUNDLE_CONFIGMAP_LABEL: "true"}).AsSelector()
+		err = r.client.List(context.TODO(), &client.ListOptions{LabelSelector: selector}, configMapList)
+		if err != nil {
+			log.Println(err)
+			return reconcile.Result{}, err
+		}
+		configMapsToChange = configMapList.Items
+	} else {
+		//changing a single labeled configmap
+
+		//get the requested object
+		requestedCAbundleConfigMap := &corev1.ConfigMap{}
+		requestedCAbundleConfigMapName := types.NamespacedName{
+			Namespace: request.Namespace,
+			Name:      request.Name,
+		}
+		err = r.client.Get(context.TODO(), requestedCAbundleConfigMapName, requestedCAbundleConfigMap)
+		if err != nil {
+			log.Println(err)
```

Comment:
Subtle case: you might have a bit of a stale notification, and the ConfigMap has since been deleted. So, you need to add
```
if apierrors.IsNotFound(err){
  return reconcile.Result{}, nil
}
return reconcile.Result{}, err
```

Reply:
fixed